Submodule updates for Bizhawk new callback system

VelpaChallenger opened the pull request:
If the Lua callback doesn't return anything (nil), the same unmodified value will be used by the core

VelpaChallenger committed:
update memory hooks to receive value from callback

If the Lua callback doesn't return anything (nil), the same unmodified value will be used by the core

vadosnaprimer commented on core/m68k/m68kcpu.h:
```diff
@@ -899,8 +899,9 @@ INLINE uint m68ki_read_32(uint address)
-    cpu_hook(HOOK_M68K_R, 4, address, val);
+  if (UNLIKELY(cpu_hook)) {
+    val = cpu_hook(HOOK_M68K_R, 4, address, val);
+  }
```
Why the brackets suddenly?

VelpaChallenger replied:
Leftover from the original twice reading code. Removed.

vadosnaprimer commented on core/debug/cpuhook.h:
```diff
@@ -43,6 +43,8 @@
 #define _CPUHOOK_H_
 
+#include "types.h"
+
```
Do you really need `types.h` for `unsigned int`?

VelpaChallenger replied:
No, that was when I was trying uint32. Removed.

VelpaChallenger committed:
remove types.h and brackets

types.h is not necessary for unsigned int and brackets was a leftover from twice reading code

vadosnaprimer commented on core/debug/cpuhook.h:
```diff
@@ -43,7 +43,6 @@
 #ifndef _CPUHOOK_H_
 #define _CPUHOOK_H_
 
-
```
Please undo this line too.

VelpaChallenger replied:
Undone.

VelpaChallenger committed:
restore line difference

Morilli commented:
We could make this support executes too (`HOOK_M68K_E`). Apparently the current code doesn't even pass any value for executes (like the opcode) because it's called before the instruction is fetched.
